megamek/src/megamek/client/ui/swing/ClientGUI.java

```diff
@@ -2129,7 +2129,16 @@ public void gameClientFeedbackRquest(GameCFREvent evt) {
                     }
                     break;
                 case Packet.COMMAND_CFR_TELEGUIDED_TARGET:
-                    List<String> targetDescriptions = evt.getTelemissileTargetDescriptions();
+                    List<Integer> targetIds = evt.getTelemissileTargetIds();
+                    List<Integer> toHitValues = evt.getTmToHitValues();
+                    List<String> targetDescriptions = new ArrayList<String>();
+                    for (int id : targetIds) {
+                        Entity tgt = client.getGame().getEntity(id);
+                        int th = toHitValues.get(targetIds.indexOf(id));
```

Comment:
I presume `targetIds` isn't large? In any case, to avoid loops-in-loops (indexOf causes that) prefer logic like:
```java
for (int ii = 0; ii < targetIds.size(); ++ii) {
  int id = targetIds.get(ii);
  int th = toHitValues.get(ii);
  Entity tgt = client.getGame().getEntity(id);
  // ... same as before
}
```

Reply:
Fixed.